L1Trigger/TrackFindingTracklet/src/Sector.cc

```diff
@@ -0,0 +1,486 @@
+    MemoriesV_[i]->clean();
+  }
+}
+
+void Sector::executeVMR() {
+  if (settings_->writeMonitorData("IL")) {
+    ofstream& out = globals_->ofstream("inputlink.txt");
+    for (unsigned int i = 0; i < IL_.size(); i++) {
+      out << IL_[i]->getName() << " " << IL_[i]->nStubs() << endl;
+    }
+  }
+
+  for (unsigned int i = 0; i < VMR_.size(); i++) {
+    VMR_[i]->execute();
+  }
+}
+
+void Sector::executeTE() {
+  for (unsigned int i = 0; i < TE_.size(); i++) {
+    TE_[i]->execute();
+  }
+}
+
+void Sector::executeTED() {
+  for (unsigned int i = 0; i < TED_.size(); i++) {
+    TED_[i]->execute();
+  }
+}
+
+void Sector::executeTRE() {
+  for (unsigned int i = 0; i < TRE_.size(); i++) {
+    TRE_[i]->execute();
+  }
+}
+
+void Sector::executeTP() {
+  for (unsigned int i = 0; i < TP_.size(); i++) {
+    TP_[i]->execute();
+  }
+}
+
+void Sector::executeTC() {
+  for (unsigned int i = 0; i < TC_.size(); i++) {
+    TC_[i]->execute();
+  }
+
+  if (settings_->writeMonitorData("TrackProjOcc")) {
+    ofstream& out = globals_->ofstream("trackprojocc.txt");
+    for (unsigned int i = 0; i < TPROJ_.size(); i++) {
+      out << TPROJ_[i]->getName() << " " << TPROJ_[i]->nTracklets() << endl;
+    }
+  }
+}
+
+void Sector::executeTCD() {
+  for (unsigned int i = 0; i < TCD_.size(); i++) {
+    TCD_[i]->execute();
+  }
+}
+
+void Sector::executePR() {
+  for (unsigned int i = 0; i < PR_.size(); i++) {
+    PR_[i]->execute();
+  }
+}
+
+void Sector::executeME() {
+  for (unsigned int i = 0; i < ME_.size(); i++) {
+    ME_[i]->execute();
+  }
+}
+
+void Sector::executeMC() {
+  for (unsigned int i = 0; i < MC_.size(); i++) {
+    MC_[i]->execute();
+  }
+}
+
+void Sector::executeMP() {
+  for (unsigned int i = 0; i < MP_.size(); i++) {
+    MP_[i]->execute();
+  }
+}
+
+void Sector::executeFT() {
+  for (unsigned int i = 0; i < FT_.size(); i++) {
+    FT_[i]->execute();
+  }
+}
+
+void Sector::executePD(std::vector<Track*>& tracks) {
+  for (unsigned int i = 0; i < PD_.size(); i++) {
+    PD_[i]->execute(tracks);
+  }
+}
+
+bool Sector::foundTrack(ofstream& outres, L1SimTrack simtrk) {
+  bool match = false;
+  for (unsigned int i = 0; i < TF_.size(); i++) {
+    if (TF_[i]->foundTrack(outres, simtrk))
+      match = true;
+  }
+  return match;
+}
+
+std::vector<Tracklet*> Sector::getAllTracklets() {
+  std::vector<Tracklet*> tmp;
+  for (unsigned int i = 0; i < TPAR_.size(); i++) {
+    for (unsigned int j = 0; j < TPAR_[i]->nTracklets(); j++) {
+      tmp.push_back(TPAR_[i]->getFPGATracklet(j));
+    }
+  }
+  return tmp;
+}
+
+std::vector<std::pair<Stub*, L1TStub*> > Sector::getStubs() const {
+  std::vector<std::pair<Stub*, L1TStub*> > tmp;
+
+  for (unsigned int imem = 0; imem < IL_.size(); imem++) {
+    for (unsigned int istub = 0; istub < IL_[imem]->nStubs(); istub++) {
+      tmp.push_back(IL_[imem]->getStub(istub));
+    }
+  }
+
+  return tmp;
+}
+
+std::set<int> Sector::seedMatch(int itp) {
+  std::set<int> tmpSeeds;
```

Comment:
does this need to be sorted? if not, use unordered_set

Reply:
no, changed to unordered set